src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1988,22 +1988,31 @@ void Unit::AttackerStateUpdate(Unit* victim, WeaponAttackType attType, bool extr
         m_currentSpells[CURRENT_MELEE_SPELL]->cast();
     else
     {
-        // attack can be redirected to another target
-        victim = GetMeleeHitRedirectTarget(victim);
+        std::list<AuraEffect*> meleeAttackOverrides = GetAuraEffectsByType(SPELL_AURA_OVERRIDE_AUTOATTACK_WITH_MELEE_SPELL);
+        if (!meleeAttackOverrides.empty())
+        {
+            for (AuraEffect* effect : meleeAttackOverrides)
+                CastSpell(victim, effect->GetSpellEffectInfo()->TriggerSpell, true, nullptr, effect);
```

Comment:
there are a few spells with a misc value + trigger spell. Trigger spellids seem to be used for mainhand & misc value for offhand. Sample spell: 204151 sniff ref: https://gist.github.com/mdX7/ae223f244cc92f10e678453e401e6768